### PR TITLE
(PUBLISH 9/25 - check with SME before publishing) Add documentation for streaming export compression

### PR DESCRIPTION
**Pull request by ecchilds**
## Give us some context

* Jira ticket: NR-139772
* This updates the documentation for streaming export to include the new compression flag, as well as automatic decompression options.
* The streaming export compression feature is **not live yet**, and until it is, **DO NOT MERGE**.

### CODE REVIEW

**Review by homelessbirds — DISMISSED** [review]

**Comment by akristen** [comment]
Duplicate PR from this: https://github.com/newrelic/docs-website/pull/14279

Was accidentally merged, but the content itself has already been reviewed and approved. We're waiting on an official release date and need confirmation from @ecchilds before merging. :)

**Comment by rhetoric101** [comment]
Check with @ecchilds for an update about the status during the week of 9/11/2023.

**Comment by ally-sassman** [comment]
@ecchilds will let us know when it's time to merge this PR.

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:415]:
```suggestion
Notably, when compression is enabled on a pre-existing export rule, it's possible for the next batch to contain both compressed and decompressed data. This has been observed in AWS, and is caused by batching within Kinesis Firehose, though it may be present in Azure as well. This can be avoided by temporarily disabling the export rule before enabling compression, or by creating a new, separate Kinesis Firehose stream for the compressed data to flow through. If this does happen to you, it is possible to view the compressed part of the data by:
1. Manually downloading the object.
2. Separating it into two separate files by copy-pasting the compressed data into a new file.
3. Decompressing the new, compressed-only data file.

If desired, you can then re-upload the two separate files back to S3 (or whatever other service you're using) and delete the old file.

```

Comment (ally-sassman) [reply src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:415]: @ecchilds How does this revision look?

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:16]:
```suggestion
As data is ingested by your New Relic organization, our streaming export feature sends that data to an AWS Kinesis Firehose or Azure Event Hub. You can set up custom rules, which are defined using [NRQL](/docs/query-your-data/nrql-new-relic-query-language/get-started/introduction-nrql-new-relics-query-language), that govern what kinds of New Relic data you'll export. You can also elect to have this data compressed before it's exported, using our new [Export Compression](#compression) feature.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:399]:
```suggestion
## Export compression [#compression]
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:406]:
```suggestion
Optionally, you can compress payloads before they are exported to save money on egress COGS. Compression is disabled by default. This can help avoid hitting your ingested data limit and save money on egress COGS.

You can enable compression using the `payloadCompression` field under `ruleParameters`. This field can be any of the following values:

* `DISABLED`: Payloads will not be compressed before being exported. If unspecified, `payloadCompression` will default to this value.
* `GZIP`: Compress payloads with the GZIP format before exporting
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:434]:
```suggestion
       * The **Supporting Access Point** on this Lambda access point will need to be set to the access point you set up on the S3 bucket.
       * Under **Transformation Configuration**:
       * The **GetObject** box must be checked.
       * The DecompressGZFunction Lambda function (or whichever other function is necessary, if a different compression format is used) must be specified.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:425]:
```suggestion
  <Collapser id="collapser-1" title="Use object metadata access points (Lambda)">
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:437]:
```suggestion
  <Collapser id="collapser-2" title="Use our Lambda metadata setting function">
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:442]:
```suggestion
    * This role must have a policy with these permissions: `s3:PutObject` and `s3:GetObject`.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:443]:
```suggestion
    3. In AWS, navigate to the [Lambda functions page](https://console.aws.amazon.com/lambda/home#/functions).
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:444]:
```suggestion
    4. Click **Create function**.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:446]:
```suggestion
    6. Click **Change default execution role >  Use an existing role**. Enter the role you created in step 2 here.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:447]:
```suggestion
    7. Scroll down and click **Create function**.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:448]:
```suggestion
    8. Once the function has been created, click **Upload from** and select **.zip or .jar file** from the dropdown.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:451]:
```suggestion
    11. All that's left to do now is enable this lambda function to trigger S3 object creation. Click **Add trigger** to start setting that up.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:457]:
```suggestion
    The Lambda function will now start automatically adding the compression metadata to all newly added S3 objects.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:467]: @ecchilds Is this in a UI? If so, it should be bolded, and include a preface (i.e., "In AWS, navigate to **Inputs**...")

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:473]:
```suggestion
4. Click **Save** at the bottom of the pane.
5. Click **Query** on the side of the screen.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ally-sassman) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:474]:
```suggestion
 Using the Input preview tab, you should now be able to query the event hub from this screen.
```

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ecchilds) [reply src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:451]: This would unfortunately change the meaning of the sentence, we want the Lambda function to be triggered by S3 object creation, not to trigger it itself. Maybe `to be triggered by S3 object creation` would be more clear?

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ecchilds) [reply src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:467]: Alright, I'll put together another commit to make that change.

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ecchilds) [reply src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx:406]: If we make this change, we should move the `Compression is disabled by default.` sentence to the end of the paragraph for the sake of clarity. We might also want to remove the redundant `and save money on egress COGS` at the end.

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx
Comment (ecchilds) [reply src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx]: Done.

**Review by ghost — APPROVED** [review]

**Pull request merged**
